src/unbeheader/cli.py

```diff
@@ -80,7 +80,14 @@ def _run_on_repo(year: int, ci: bool) -> bool:
         print(cformat('Updating headers to the year %{yellow!}{year}%{reset} for all '
                       'git-tracked files...').format(year=year))
     try:
-        for file_path in subprocess.check_output(['git', 'ls-files'], text=True).splitlines():
+        # Get all files tracked by git
+        git_file_paths = set(subprocess.check_output(['git', 'ls-files'], text=True).splitlines())
+        # Exclude deleted files
+        git_file_paths -= set(subprocess.check_output(['git', 'ls-files', '--deleted'], text=True).splitlines())
```

Comment:
I'd do this last. Dunno if we can get a case where the next one includes something unwanted but since you never want deleted files in the set...

Reply:
Done.